helix-term/src/commands.rs

```diff
@@ -4559,9 +4559,14 @@ fn shell_impl(
         .stdout(Stdio::piped())
         .stderr(Stdio::piped());
 
+    #[cfg(not(target_family = "windows"))]
     if input.is_some() {
         process.stdin(Stdio::piped());
     }
+    #[cfg(target_family = "windows")]
+    {
+        process.stdin(Stdio::piped());
+    }
```

Comment:
This doesn't need conditional compiling since it compiles ok on and off windows. The runtime behavior should be influenced by it though and that can be checked with the [`cfg!`](https://doc.rust-lang.org/rust-by-example/attribute/cfg.html) macro

```rs
if input.is_some() || cfg!(windows) {
    precss.stdin(Stdio::piped());
}
```

Reply:
Is this usually more preferable?

Reply:
From what I understand, it's best to only use what conditional compilation you really need.  That way you don't mistakenly end up with wildly different code-paths on different operating systems. At the very least it makes rust-analyzer a little quieter to keep conditional compiles to a minimum :P

Reply:
Gotcha, gotcha. Thanks!

I did have an issue with my first attempt, even though I was pretty sure it wasn't going to work. It _did_ compile, but the binary would just hang there. The most surprising thing is that this was legal, and it compiled.

```rust
#[cfg(target_family = "windows")]
let a = "hello";
```